### DISCUSSION TITLE
Type casting in typecheck

### DISCUSSION THREAD

**Discussion by algo-robin** [agronholm/typeguard#268] [Ideas]
I would love for `@typechecked` to include type casting similar to how I can give '1' to an int Field in pydantic and have it spit out an integer 1; possibly enabled by a parameter for folks who want to stick to raising a TypeError in the above mentioned case.

Is this something that aligns with the project and would be possible to implement?

**Comment by agronholm** [comment]
I'm afraid this is out of scope for this project.

**Reply by agronholm** [reply]
The current `master` version can actually do `value = check_type(some_function(), YourType)` and as a result `value` should be seen by static type checkers as `YourType`, in addition of getting checked at run time. Is this what you wanted?

**Reply by agronholm** [reply]
@robinderat Does the `check_type()` function in v3.x do what you wanted? If so, please mark this as answered.

**Reply by algo-robin** [reply]
Not quite,

Current behaviour is
```
from typeguard import check_type 
a = check_type('1', int)
-------
typeguard.TypeCheckError: str is not an instance of int
```

whereas i was hoping for 
```
from typeguard import check_type
a = check_type('1', int)
print(a, type(a))
-----
1, int
```

**Reply by agronholm** [reply]
How would that even work with arbitrary types?
